src/systemcmds/tests/test_adc.cpp

```diff
@@ -37,56 +37,36 @@
  */
 
 #include <px4_platform_common/time.h>
-#include <px4_platform_common/px4_config.h>
-#include <px4_platform_common/posix.h>
 #include <px4_platform_common/log.h>
-
-#include <sys/types.h>
-
-#include <stdio.h>
-#include <stdlib.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-
 #include "tests_main.h"
-
 #include <drivers/drv_adc.h>
 
+#include <uORB/Subscription.hpp>
+#include <uORB/topics/adc_report.h>
+
 int test_adc(int argc, char *argv[])
 {
-	int fd = px4_open(ADC0_DEVICE_PATH, O_RDONLY);
-
-	if (fd < 0) {
-		PX4_ERR("ERROR: can't open ADC device");
-		return 1;
-	}
-
-	for (unsigned i = 0; i < 5; i++) {
-		/* make space for a maximum number of channels */
-		px4_adc_msg_t data[PX4_MAX_ADC_CHANNELS];
-		/* read all channels available */
-		ssize_t count = px4_read(fd, data, sizeof(data));
+	uORB::Subscription	_adc_sub{ORB_ID(adc_report)};
+	adc_report_s adc;
 
-		if (count < 0) {
-			goto errout_with_dev;
-		}
+	px4_usleep(1000000);	// sleep 1s and wait for adc report
```

Comment:
We shouldn't wait that long, it just delays tests.

Reply:
Done.